src/decoder/mod.rs

```diff
@@ -1,18 +1,40 @@
 //! Decodes samples from an audio file.
 
-use std::error::Error;
+#[cfg(all(feature = "symphonia-mp3", feature = "mp3"))]
+compile_error!("Both symphonia-mp3 and mp3 are enabled. 
```

Comment:
Please don't add these compiler errors. Cargo features are designed to be additive and some distros enable *all* features of a crate. This causes headaches for them. This case can be handled gracefully in the code by always preferring one backend if both are enabled.

Reply:
Makes sense, I was thinking people would rather not have duplicate dependencies enabled to reduce binary size, but I can try to use the docs for that instead.